pytest_fixtures/certificate.py

```diff
@@ -0,0 +1,150 @@
+from pathlib import Path
+
+import pytest
+from fauxfactory import gen_string
+
+from robottelo import ssh
+from robottelo.config import settings
+from robottelo.helpers import get_data_file
+from robottelo.ssh import get_connection
+from robottelo.ssh import upload_file
+
+
+@pytest.fixture(scope="module")
+def cert_data():
+    """Get host name, scripts, and create working directory."""
+
+    def _cert_data(hostname=None):
+        sat6_hostname = hostname or settings.server.hostname
+        capsule_hostname = 'capsule.example.com'
+        key_file_name = '{0}/{0}.key'.format(sat6_hostname)
+        cert_file_name = '{0}/{0}.crt'.format(sat6_hostname)
+        ca_bundle_file_name = 'cacert.crt'
+        success_message = 'Validation succeeded'
+        cert_data = {
+            'sat6_hostname': sat6_hostname,
+            'capsule_hostname': capsule_hostname,
+            'key_file_name': key_file_name,
+            'cert_file_name': cert_file_name,
+            'ca_bundle_file_name': ca_bundle_file_name,
+            'success_message': success_message,
+        }
+        return cert_data
+
+    return _cert_data
+
+
+@pytest.fixture()
+def cert_setup(cert_data):
+    """copy all configuration files to satellite host for generating custom certs"""
+
+    def _cert_setup(hostname=None):
+        # Need a subdirectory under ssl-build with same name as Capsule name
+        hostname = hostname or settings.server.hostname
```

Comment:
I really think you should be using `robottelo.Satellite` class instances here, and its `execute` method to send these ssh commands, instead of importing `get_connection`. Changing that pattern will move all of this ssh setup to ssh2-python through broker, which is far faster. It also gets rid of the need to use all of these context managers